test/extension-tests/successful-build/test/variable-substitution.test.ts

```diff
@@ -0,0 +1,184 @@
+import * as chai from 'chai';
+import {expect} from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+chai.use(chaiAsPromised);
+
+import {clearExistingKitConfigurationFile} from '../../../test-helpers';
+import {DefaultEnvironment} from '../../../helpers/test/default-environment';
+
+import * as api from '../../../../src/api';
+import * as path from 'path';
+import {CMakeCache} from '../../../../src/cache';
+import {CMakeTools} from '../../../../src/cmake-tools';
+import {normalizePath} from '../../../../src/util';
+
+suite('[Variable Substitution]', async () => {
+  let cmt: CMakeTools;
+  let testEnv: DefaultEnvironment;
+
+  setup(async function(this: Mocha.IBeforeAndAfterContext) {
+    if (process.env.HasVs != 'true') {
+      this.skip();
+    }
+    this.timeout(100000);
+
+    testEnv = new DefaultEnvironment('test/extension-tests/successful-build/project-folder');
+    cmt = await CMakeTools.create(testEnv.vsContext);
+
+    // This test will use all on the same kit.
+    // No rescan of the tools is needed
+    // No new kit selection is needed
+    await clearExistingKitConfigurationFile();
+    await cmt.scanForKits();
+    await cmt.selectKit();
+
+    testEnv.projectFolder.buildDirectory.clear();
+  });
+
+  teardown(async function(this: Mocha.IBeforeAndAfterContext) {
+    this.timeout(30000);
+    await cmt.asyncDispose();
+    testEnv.teardown();
+  });
+
+  test('Check substitution for "workspaceRoot"', async () => {
+    // Set fake settings
+    testEnv.setting.changeSetting('configureSettings', {workspaceRoot: '${workspaceRoot}'});
```

Comment:
This settings is missing in the new documentation. Please add it there.

https://github.com/vector-of-bool/vscode-cmake-tools/blob/feature/refactor-1.0/docs/configuring.rst

Reply:
I will add it after I have fixed my current issues with `xxxEnvironment`.

Reply:
Just on the note of documentation.. atm A LOT is missing from it.

Reply:
Done.